### ISSUE TITLE
[BUG] Fix status error returned from rest code

### ISSUE THREAD

**Issue by gnufied** [vmware/govmomi#2521]
**Describe the bug**

When creating tags or cateogies, the sdk returns following error type:

https://github.com/vmware/govmomi/blob/master/vapi/rest/client.go#L178

```
return &statusError{res}
```

The problem is - users of library generally want to know the http error code at very minimum, so as they can make reasonable assumption about error. Such as - did getting a resource failed because of network error or was it 404?

**Comment by embano1** [comment]
@dougm this is also something I think we should address mid-term. Not sure about the impact on the APIs, thoughts?

**Comment by dougm** [comment]
Agreed, I had started to look at doing this a while back.. was thinking we could export the existing `soap.statusError` and also expose the `http.Response.StatusCode`.  Then use that error type in the rest client too.

https://github.com/vmware/govmomi/blob/4a7de71a1d515a62ecf5a62c043e7e3859359b13/vim25/soap/client.go#L560-L584

**Comment by brakthehack** [comment]
Interested in seeing this come about. A few more options:

1. Make statusError public and have it carry core parts of the request such status code. If we expose it, then clients can at least cast to that error and grab the response information when they need to.
2. Optionally, update all signatures to return this error.
3. For operations that return an object with a pointer, make the pointer nil when 404 is encountered. Seems there are similar issues which report that this should be expected behavior: https://github.com/vmware/govmomi/issues/1639.
